### PR TITLE
Update azuresearch.py

### PR DESCRIPTION
**Pull request by shubhamgupta568**
This change has been done to fix the below error.   File ~\Anaconda3\envs\dd_original\lib\site-packages\langchain_community\vectorstores\azuresearch.py:628, in <listcomp>(.0)
    592     semantic_answers_dict[semantic_answer.key] = {
    593         "text": semantic_answer.text,
    594         "highlights": semantic_answer.highlights,
    595     }
    596 # Convert results to Document objects
    597 docs = [
    598     (
    599         Document(
    600             page_content=result.pop(FIELDS_CONTENT),
    601             metadata={
    602                 **(
    603                     json.loads(result[FIELDS_METADATA])
    604                     if FIELDS_METADATA in result
    605                     else {
    606                         k: v
    607                         for k, v in result.items()
    608                         if k != FIELDS_CONTENT_VECTOR
    609                     }
    610                 ),
    611                 **{
    612                     "captions": {
    613                         "text": result.get("@search.captions", [{}])[0].text,
    614                         "highlights": result.get("@search.captions", [{}])[
    615                             0
    616                         ].highlights,
    617                     }
    618                     if result.get("@search.captions")
    619                     else {},
    620                     "answers": semantic_answers_dict.get(
    621                         result.get(FIELDS_ID, ""),
    622                         "",
    623                     ),
    624                 },
    625             },
    626         ),
    627         float(result["@search.score"]),
--> 628         float(result["@search.reranker_score"]),
    629     )
    630     for result in results
    631 ]
    632 return docs

TypeError: float() argument must be a string or a number, not 'NoneType'

Thank you for contributing to LangChain!

- [ ] **PR title**: "package: description"
  - Where "package" is whichever of langchain, community, core, experimental, etc. is being modified. Use "docs: ..." for purely docs changes, "templates: ..." for template changes, "infra: ..." for CI changes.
  - Example: "community: add foobar LLM"

- [ ] **PR message**: ***Delete this entire checklist*** and replace with
    - **Description:** a description of the change
    - **Issue:** the issue # it fixes, if applicable
    - **Dependencies:** any dependencies required for this change
    - **Twitter handle:** if your PR gets announced, and you'd like a mention, we'll gladly shout you out!

- [ ] **Add tests and docs**: If you're adding a new integration, please include
  1. a test for the integration, preferably unit tests that do not rely on network access,
  2. an example notebook showing its use. It lives in `docs/docs/integrations` directory.

- [ ] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified. See contribution guidelines for more: https://python.langchain.com/docs/contributing/

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in langchain.

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, hwchase17.

### CODE REVIEW

File: libs/community/langchain_community/vectorstores/azuresearch.py [hunk 1]
```diff
@@ -625,7 +625,7 @@ def semantic_hybrid_search_with_score_and_rerank(
                     },
                 ),
                 float(result["@search.score"]),
-                float(result["@search.reranker_score"]),
+                float(result["@search.reranker_score"]) if result["@search.reranker_score"] else None,
```
Comment (ccurme) [inline libs/community/langchain_community/vectorstores/azuresearch.py:628]: In what circumstances will the score be None?

**Comment by ccurme** [comment]
@lawadam would you mind taking a look?

File: libs/community/langchain_community/vectorstores/azuresearch.py (hunk 1 above)
Comment (lawadam) [reply libs/community/langchain_community/vectorstores/azuresearch.py:628]: Do you have a test case that shows this behavior? I haven't seen it be None before.

Comment (shubhamgupta568) [reply libs/community/langchain_community/vectorstores/azuresearch.py:628]: Hello @lawadam and  @ccurme , I’m currently involved in the Adaptive RAG agent process at Ernst & Young. Our first step is to classify the query. Following that, we utilize Azure Search in conjunction with a vector store for asimilarity_search of the “semantic_hybrid” type. However, we’re encountering an error in this specific instance. Due to the proprietary nature of my organization’s work, I’m unable to share the code or test case with you.

**Comment by hwchase17** [comment]
needs MRE

**Comment by teebu** [comment]
This error keeps coming up in my Azure AI Studio deployment. Very annoying to encounter. I don't know what's causing it, seems to be long query. I can't provide more details.

**Pull request closed without merging**